src/cargo/ops/cargo_add/mod.rs

```diff
@@ -196,6 +196,14 @@ pub fn add(workspace: &Workspace<'_>, options: &AddOptions<'_>) -> CargoResult<(
         print_dep_table_msg(&mut options.config.shell(), &dep)?;
 
         manifest.insert_into_table(&dep_table, &dep)?;
+        if let Some(option) = dep.optional {
+            let rust_version_check =
+                check_rust_version_for_optional_dependency(workspace.rust_version())?;
```

Comment:
We should be checking the rust-version of the package being modified, rather than the lowest common denominator of the workspace.

Reply:
thanks and solved.